UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/model/admission/ExamAdmission.java

```diff
@@ -92,7 +93,24 @@ private String toIndentedString(Object o) {
         AdmissionContractUtil cUtil = new AdmissionContractUtil();
         ArrayList<InvalidParameter> invalidParameters = super.getSemanticErrors(stub);
 
-        // TODO add remaining checks once all necessary contracts available
+        if (!cUtil.checkExamExists(stub, this)) {
+            invalidParameters.add(cUtil.getAdmissionExamNotExistsParam());
+        }
+        if (!cUtil.checkExamAvailableForStudent(stub, this)) {
+            invalidParameters.add(cUtil.getAdmissionExamNotAvailableParam("enrollmentId"));
+            invalidParameters.add(cUtil.getAdmissionExamNotAvailableParam("examId"));
+        }
+        if (!cUtil.checkExamAdmissionNotAlreadyExists(stub, this)) {
+            invalidParameters.add(cUtil.getAdmissionAlreadyExistsParam("enrollmentId"));
+            invalidParameters.add(cUtil.getAdmissionAlreadyExistsParam("examId"));
+        }
+        if (!cUtil.checkCourseAdmissionExists(stub, this)) {
+            invalidParameters.add(cUtil.getCourseAdmissionNotExistsParam("enrollmentId"));
+            invalidParameters.add(cUtil.getCourseAdmissionNotExistsParam("examId"));
+        }
+        if (!cUtil.checkExamAdmittable(stub, this)) {
+            invalidParameters.add(cUtil.getAdmissionNotPossibleParam());
+        }
```

Comment:
What happens for e.g. empty examId? Will we get both the error that it should not be empty, and every semantic error related to not finding the exam?

Reply:
yeah, I am not satisfied with that  - imho we should only submit the exact error (only empty)  but that increases the complexity. I guess we can deal with this kind of "better error management" later on, once all the functionality is established

Reply:
https://github.com/upb-uc4/hlf-chaincode/issues/194